indexer/utils/abi_setting.py

```diff
@@ -0,0 +1,212 @@
+        ],
+        "name": "Withdrawal",
+        "type": "event",
+    }
+)
+
+transfer_event = Event(
+    {
+        "anonymous": False,
+        "inputs": [
+            {"indexed": True, "name": "from", "type": "address"},
+            {"indexed": True, "name": "to", "type": "address"},
+            {"indexed": False, "name": "value", "type": "uint256"},
+        ],
+        "name": "Transfer",
+        "type": "event",
+    }
+)
+
+single_transfer_event = Event(
+    {
+        "anonymous": False,
+        "inputs": [
+            {"indexed": True, "name": "operator", "type": "address"},
+            {"indexed": True, "name": "from", "type": "address"},
+            {"indexed": True, "name": "to", "type": "address"},
+            {"indexed": False, "name": "id", "type": "uint256"},
+            {"indexed": False, "name": "value", "type": "uint256"},
+        ],
+        "name": "TransferSingle",
+        "type": "event",
+    }
+)
+
+batch_transfer_event = Event(
+    {
+        "anonymous": False,
+        "inputs": [
+            {"indexed": True, "name": "operator", "type": "address"},
+            {"indexed": True, "name": "from", "type": "address"},
+            {"indexed": True, "name": "to", "type": "address"},
+            {"indexed": False, "name": "ids", "type": "uint256[]"},
+            {"indexed": False, "name": "values", "type": "uint256[]"},
+        ],
+        "name": "TransferBatch",
+        "type": "event",
+    }
+)
+
+# ABI function
+NAME_ABI_FUNCTION = Function(
+    {
+        "constant": True,
+        "inputs": [],
+        "name": "name",
+        "outputs": [{"name": "", "type": "string"}],
+        "payable": False,
+        "stateMutability": "view",
+        "type": "function",
+    }
+)
+
+SYMBOL_ABI_FUNCTION = Function(
+    {
+        "constant": True,
+        "inputs": [],
+        "name": "symbol",
+        "outputs": [{"name": "", "type": "string"}],
+        "payable": False,
+        "stateMutability": "view",
+        "type": "function",
+    }
+)
+
+DECIMALS_ABI_FUNCTION = Function(
+    {
+        "constant": True,
+        "inputs": [],
+        "name": "decimals",
+        "outputs": [{"name": "", "type": "uint8"}],
+        "payable": False,
+        "stateMutability": "view",
+        "type": "function",
+    }
+)
+
+TOTAL_SUPPLY_ABI_FUNCTION = Function(
+    {
+        "constant": True,
+        "inputs": [],
+        "name": "totalSupply",
+        "outputs": [{"name": "", "type": "uint256"}],
+        "payable": False,
+        "stateMutability": "view",
+        "type": "function",
+    }
+)
+
+TOTAL_SUPPLY_WITH_ID_ABI_FUNCTION = Function(
+    {
+        "constant": True,
+        "inputs": [{"name": "id", "type": "uint256"}],
+        "name": "totalSupply",
+        "outputs": [{"name": "", "type": "uint256"}],
+        "payable": False,
+        "stateMutability": "view",
+        "type": "function",
+    }
+)
+
+TOKEN_SUPPLY_ABI_FUNCTION = Function(
```

Comment:
Where does this come from?

Reply:
I can't remember where I saw this abi function, it doesn't seem to be in use right now, I'll remove this code in a subsequent commit